docs/modules/rag.mdx

```diff
@@ -0,0 +1,181 @@
+    api_key=os.getenv("WATSONX_APIKEY"),
+    base_url=os.getenv("WATSONX_URL"),
+)
+
+# BeeAI native vector store
+vector_store = VectorStore.from_name(
+    name="beeai:TemporalVectorStore", 
+    embedding_model=embedding_model
+)
+```
+
+<Tip>
+Native BeeAI modules can be loaded directly by importing and instantiating the module, e.g. `from beeai_framework.adapters.beeai.backend.vector_store import TemporalVectorStore`.
+</Tip>
+
+### Supported Provider's Vector Store
+
+```python
+# LangChain integration
+vector_store = VectorStore.from_name(
+    name="langchain:InMemoryVectorStore", 
+    embedding_model=embedding_model
+)
+```
+
+<Tip>
+You can customize dynamically loaded components by passing additional parameters directly to the `from_name` method. These parameters will be forwarded to the component's constructor, allowing you to configure settings like batch sizes, connection pools, or other provider-specific options without changing your code structure.
+</Tip>
+
+## RAG Agent
+
+The RAG Agent implements a sophisticated retrieval-augmented generation pipeline that combines the power of semantic search with large language models. The agent follows a three-stage process and supports advanced configuration options including custom reranking, flexible retrieval parameters, comprehensive error handling, and query flexibility using various object types.
+
+### 1. Retrieval
+The agent searches the vector store using semantic similarity to find the most relevant documents for the user's query. You can configure the number of documents retrieved and similarity thresholds to optimize for your specific use case.
+
+### 2. Reranking (Optional)
+Retrieved documents can be reranked using advanced LLM-based models to improve relevance and quality of the context provided to the generation stage. This step significantly enhances response accuracy for complex queries.
+
+### 3. Generation
+The LLM generates a response using the retrieved documents as context, ensuring grounded and accurate answers. Built-in error handling ensures informative error messages are stored in memory when issues occur.
+
+### Basic Usage
+
+<Note>
+Document loading and population of the vector store is the developers's responsibility and out of scope for the agent. 
+</Note>
+
+<CodeGroup>
+
+```py Python
+
+# Create vector store (assumes documents are already populated)
+vector_store = VectorStore.from_name(
+    name="...", 
+    embedding_model=...
+)
+
+# Initialize chat model and reranker
+llm = ChatModel.from_name("ollama:llama3.2:latest")
+reranker = LLMDocumentReranker(llm, top_n=3)
+
+# Create RAG agent
+agent = RAGAgent(
+    llm=llm,
+    memory=UnconstrainedMemory(),
+    vector_store=vector_store,
+    reranker=reranker,
+    number_of_retrieved_documents=5
+)
+
+# Run the agent
+response = await agent.run(
+    RagAgentRunInput(message=UserMessage("What agents are available in BeeAI?"))
+)
+print(response.message.text)
```

Comment:
I haven't noticed that you changed the `run` method interface to take a class instead of keyword arguments. The `run` method is now quite verbose. Unfortunately.